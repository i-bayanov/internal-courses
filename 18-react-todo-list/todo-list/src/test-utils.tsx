import React from 'react';
import { render } from 'react-dom';
import { BrowserRouter } from 'react-router-dom';

import { Provider } from 'react-redux';
import { configureStore } from '@reduxjs/toolkit';

import todoReducer from './reducer';

const STORE = [
  {
    id: '1651234007958',
    title: 'Write a to-do list',
    completed: true,
  },
  {
    id: '1651234009086',
    title: 'Mark the first item as completed',
    completed: true,
  },
  {
    id: '1651234010110',
    title: '-',
    completed: false,
  },
  {
    id: '1651234864772',
    title: 'Come up with an interesting third point',
    completed: false,
  },
];

const mockStore = configureStore({ reducer: todoReducer, preloadedState: STORE });

// eslint-disable-next-line no-undef
const testRender = (component: JSX.Element, container: Element) => render(
  <Provider store={mockStore}>
    <BrowserRouter>
      {component}
    </BrowserRouter>
  </Provider>,
  container,
);

export { STORE, testRender };
