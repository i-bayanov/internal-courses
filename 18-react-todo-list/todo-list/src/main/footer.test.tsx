import React from 'react';
import { unmountComponentAtNode } from 'react-dom';

import { testRender } from '../test-utils';
import Footer from './footer';

let container: Element | null = null;

beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container!);
  container!.remove();
  container = null;
});

describe('Footer component', () => {
  it('should renders correctly', () => {
    testRender(<Footer />, container!);
    expect(container).toMatchSnapshot();
  });
});
