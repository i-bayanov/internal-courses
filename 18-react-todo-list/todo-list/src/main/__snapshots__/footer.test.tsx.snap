// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component should renders correctly 1`] = `
<div>
  <footer
    class="footer"
  >
    <span
      class="todo-count"
    >
      2
       item
      s
       left
    </span>
    <ul
      class="filters"
    >
      <li>
        <a
          aria-current="page"
          class="selected"
          href="/"
        >
          All
        </a>
      </li>
      <li>
        <a
          class=""
          href="/active"
        >
          Active
        </a>
      </li>
      <li>
        <a
          class=""
          href="/completed"
        >
          Completed
        </a>
      </li>
    </ul>
    <button
      class="clear-completed"
      style="display: block;"
    >
      Clear completed
    </button>
  </footer>
</div>
`;
